go/libraries/doltcore/sqle/database.go

```diff
@@ -673,7 +673,15 @@ func (db Database) GetTableNames(ctx *sql.Context) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
-	return filterDoltInternalTables(tblNames), nil
+	showSystemTables, err := ctx.GetSessionVariable(ctx, dsess.ShowSystemTables)
```

Comment:
You can simplify this by creating a function which returns a boolean for if it's enabled. That function can probably swallow all errors and just return false. Up to you.